pyccel/codegen/printing/ccode.py

```diff
@@ -1487,17 +1494,48 @@ def _print_EmptyNode(self, expr):
         return ''
 
     #=================== OMP ==================
-    def _print_OMP_For_Loop(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp for{}\n{{'.format(omp_expr)
-
-    def _print_OMP_Parallel_Construct(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp {}\n{{'.format(omp_expr)
 
-    def _print_OMP_Single_Construct(self, expr):
-        omp_expr   = str(expr.txt)
-        return '#pragma omp {}\n{{'.format(omp_expr)
+    def _print_OmpCombinedAnnotatedComment(self, expr):
+        clauses = ''
+        if expr.combined:
+            clauses = ' ' + expr.combined
+        clauses += str(expr.txt)
+        omp_expr = '#pragma omp'
+        if isinstance(expr, OMP_Parallel_Construct):
+            omp_expr += ' parallel{}'.format(clauses)
+            if expr.combined is None:
+                omp_expr += '\n{'
+            elif (expr.combined and "for" not in expr.combined):
+                if "masked taskloop" not in expr.combined:
+                    omp_expr += '\n{'
+        elif isinstance(expr, OMP_Masked_Construct):
+            omp_expr += ' masked{}'.format(clauses)
+            if expr.combined is None:
+                omp_expr += '\n{'
+        elif isinstance(expr, OMP_Target_Construct):
+            omp_expr += ' target{}'.format(clauses)
+            if expr.combined is None:
+                omp_expr += '\n{'
+            elif (expr.combined and "for" not in expr.combined):
+                omp_expr += '\n{'
+        elif isinstance(expr, OMP_Teams_Construct):
+            omp_expr += ' teams{}'.format(clauses)
+            if expr.combined is None:
+                omp_expr += '\n{'
+
+        return omp_expr
+
+    def _print_OmpAnnotatedComment(self, expr):
+        omp_expr = str(expr.txt)
+        if isinstance(expr, OMP_For_Loop):
+            omp_expr = '#pragma omp for{}'.format(omp_expr)
+        else:
+            omp_expr = '#pragma omp {}'.format(omp_expr)
+        if isinstance(expr, (OMP_Sections_Construct, OMP_Section_Construct,
+                             OMP_Task_Construct, OMP_Single_Construct,
+                             OMP_Critical_Construct, OMP_Master_Construct)):
```

Comment:
Is there a property that make these cases special? (e.g. is_multiline) if so, then I think it would be better to define the property so you don't need to update this list in every implemented language whenever you want to add a new construct